_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -739,6 +750,40 @@ mfxStatus VP9CompleteProc(void *p_state, void * /* pp_param */, mfxStatus)
     return MFX_ERR_NONE;
 }
 
+mfxStatus VideoDECODEVP9_HW::ReallocInternalSurface(UMC::FrameMemID &mid, mfxFrameInfo &frameInfo)
+{
+    mfxStatus sts = MFX_ERR_NONE;
+    UMC::VideoDataInfo videoInfo;
+
+    UMC::ColorFormat const cf = GetUMCColorFormat_VP9(&m_frameInfo);
+
+    if (UMC::UMC_OK != videoInfo.Init(m_vPar.mfx.FrameInfo.Width, m_vPar.mfx.FrameInfo.Height, cf, m_frameInfo.bit_depth))
+        return MFX_ERR_MEMORY_ALLOC;
+
+#ifndef MFX_VA_LINUX
+    UMC::Status umc_sts = m_FrameAllocator->Alloc(&mid, &videoInfo, 0);
+#else
+    UMC::Status umc_sts = m_FrameAllocator->Alloc(&mid, &videoInfo, mfx_UMC_ReallocAllowed);
+    if (UMC::UMC_ERR_NOT_ENOUGH_BUFFER == umc_sts && m_adaptiveMode)
+    {
+        mfxFrameSurface1 *surf = m_FrameAllocator->GetSurfaceByIndex(mid);
+        if (!surf)
+            return MFX_ERR_INVALID_HANDLE;
+         surf->Info.Width = frameInfo.Width;
+         surf->Info.Height = frameInfo.Height;
+         VAAPIVideoCORE *vaapi_core = reinterpret_cast<VAAPIVideoCORE *>(m_core);
+         mfxStatus sts = vaapi_core->ReallocFrame(surf);
```

Comment:
redefinition of line 755. I suggest to fully remove definition at line 755 and return ```MFX_ERR_NONE``` at line 784